cmd/snap-update-ns/utils.go

```diff
@@ -654,18 +654,18 @@ func createWritableMimic(dir, neededBy string, as *Assumptions) ([]*Change, erro
 	return changes, nil
 }
 
-func applyProfile(snapName string, currentBefore, desired *osutil.MountProfile, as *Assumptions) (*osutil.MountProfile, error) {
+func applyProfile(up MountProfileUpdate, snapName string, currentBefore, desired *osutil.MountProfile, as *Assumptions) (*osutil.MountProfile, error) {
```

Comment:
will this be a method of CommonUpdateProfile in the end?

Reply:
No, it will be a function that just takes a MountProfileUpdate, it is extensively tested against a test behaviour to see what happens in specific edge cases.